### ISSUE TITLE
How to change the cell style

### ISSUE THREAD

**Issue by chuan116** [troldal/OpenXLSX#255]
How to change the cell style，For example: make text bold

**Comment by aral-matrix** [comment]
Not currently supported, as far as I know - it's on the "maybe do this, eventually" list, but I think @troldal is quite busy. For now, data, worksheet and formula manipulation is what's supported.
The styles logic of XLSX is unfortunately non-trivial. It might be possible to implement a first approach to allow the user to define a raw XML node directly with style information, that way it could at least be done programmatically.

However, styles are distributed over different XML nodes even in the styles XML file :/

**Comment by aral-matrix** [comment]
@chuan116: I have a first function complete implementation of excel styles - would you like to test it?

**Comment by chuan116** [comment]
@aral-matrix Thank you for your reply. I really want to test this new function, but how can I get the implementation of your new function

**Comment by aral-matrix** [comment]
Would it be okay for you to register on codeberg.org? I can't give read-only access to a private repository on github (thanks for nothing, Microsoft).
If you give me your codeberg.org username, I will give you read access to the current version of my branch for testing

**Comment by aral-matrix** [comment]
*bump* - @chuan116  - still interested? :)

**Comment by hmq2018** [comment]
@chuan116 I'm interested in your code, I want to try your code to change the color of cells. Could you share the code with me? Thank You!
This is my codeberg.org account: welder

**Comment by aral-matrix** [comment]
@hmq2018 you tagged the issue creator, I am the one who was offering test access - I gave you access on codeberg - please have a look at Examples/Demo10.cpp on how to use the XLStyles class / functionality. I would appreciate some feedback.

Basically, you have to manage the styles yourself, and you decide when to create a new style, and then you can assign those styles to a cell or a row or a column.

What I do *not* do, is I do not create "merged" styles in the intersection cell of a custom column style and custom row style.

Instead, when you fill a new cell with content for the first time (when it gets created), it will use the same style as the row, if any, then as the column, if no row style, or finally the default style (none)

**Comment by hmq2018** [comment]
@aral-matrix It's my fault. Fortunately, you found it. Thank you so much.
I will refer to the Examples/Demo10.cpp, Thank you again for your quick response.

**Comment by hmq2018** [comment]
@aral-matrix 
I downloaded your code today and compiled it in my environment (WIN10 + VS2019 and use C++ 17 Standard).
The process went smoothly, except for the following two places in the following files. There are errors. Maybe because of my environment, I made the following changes to make it pass.

**File: XLSharedStrings.hpp**
from:
constexpr size_t XLMaxSharedStrings = std::numeric_limits< int32_t >::max();
change to:
constexpr size_t XLMaxSharedStrings = 2147483647;

**File: XLRelationships.cpp**
from:
constexpr const size_t relFolderLen = strlen(relFolder);
change to:
constexpr const size_t relFolderLen = 6;

After compiling, I tried to modify the background color of the cell according to the example in demo10.cpp, and the actual effect was in line with my expectations.
Because I have just started learning C++, I can only solve my actual problems based on your examples. Later, I will learn and understand the principles and logic of your code implementation.

Thank you very much for your efforts!

**Comment by aral-matrix** [comment]
@hmq2018  I am glad to see this is working for you. I checked my code and the XLSharedStrings.hpp does include the required limits header file - so I am surprised that std::numeric_limits wasn't working for you.

Could you maybe open an issue on codeberg (to not confuse it with the github version) with the error messages you see when you have the numeric_limits in there? I also use the same function in XLMergeCells.hpp - and that compiled fine?

As for XLRelationships.cpp, I suspect that your compiler does not allow using strlen in a constexpr. An alternative fix you could try would be

from:
```constexpr const size_t relFolderLen = strlen(relFolder);```
change to:
```static const size_t relFolderLen = strlen(relFolder);```

basically changing the constexpr (evaluated at compile time) to a normal const expression that is initialized only once.

**Comment by easymoney322** [comment]
> so I am surprised that std::numeric_limits wasn't working for you.

That's the platform-specific issue with projects that do include Windows.h. I've mentioned it in https://github.com/troldal/OpenXLSX/issues/260 and [made a pull-request](https://github.com/troldal/OpenXLSX/pull/261) that supposed to fix this.

**Comment by aral-matrix** [comment]
I merged the development branch into main today. Issue is resolved :)
